legacy/marconi-chain-index-legacy/src/Marconi/ChainIndex/Legacy/Indexers/EpochState.hs

```diff
@@ -477,6 +483,9 @@ instance Buffered EpochStateHandle where
                     isLastImmutableBeforeRollbackable =
                       isBlockRollbackable securityParam (blockNo + 1) (getBlockNoFromChainTip chainTip)
                 when (isRollbackable || isLastImmutableBeforeRollbackable || isFirstEventOfEpoch) $ do
+                  -- The conditions above make it impossible for the ChainPoint to be at genesis,
+                  -- so we should be able to extract the slot and the hash safely.
+                  let (slotNo, blockHeaderHash) = Util.extractSlotNoAndHash chainPoint
```

Comment:
I am not 100% sure this reasoning is correct so please let me know if I'm mistaken.

Reply:
yes, we don't store anything at genesis.